lib/auth/grpcserver.go

```diff
@@ -3452,6 +3452,15 @@ func (g *GRPCServer) GenerateCertAuthorityCRL(ctx context.Context, req *proto.Ce
 	return &proto.CRL{CRL: crl}, nil
 }
 
+func (g *GRPCServer) RegisterUsingIAMMethod(srv proto.AuthService_RegisterUsingIAMMethodServer) error {
+	auth, err := g.authenticate(srv.Context())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(auth.authServer.registerUsingIAMMethod(srv))
```

Comment:
Instead of calling this on `authServer` directly, I would still make a `RegisterUsingIAMMethod` method on the `ServerWithRoles` and call it here, even if it doesn't do any RBAC checks. This way authz logic won't spread out.

Reply:
Yes, I meant to come back and do that, done in the latest revision